Sources/Commands/Workspace.swift

```diff
@@ -269,6 +271,71 @@ public class Workspace {
         return path
     }
 
+    /// Updates the current dependencies.
+    public func updateDependencies() throws {
+        let (rootManifest, _) = try loadDependencyManifests()
+        // Only create constraints based on root manifest for the update resolution.
+        let updateConstraints = computePackageConstraints(rootManifest: rootManifest, dependencies: [])
```

Comment:
Rather than in one case passing in an empty array here, I would probably just separate out the code which adds the extra "pin" constraints and simply not call it here.